File: src/main/java/vini2003/xyz/bambootweaks/registry/BambooTweaksItems.java
package vini2003.xyz.bambootweaks.registry;

import vini2003.xyz.bambootweaks.BambooTweaks;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.WallStandingBlockItem;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class BambooTweaksItems {
	public static final Item TORCH_ITEM = register("bamboo_torch", new WallStandingBlockItem(BambooTweaksBlocks.BAMBOO_TORCH_BLOCK, BambooTweaksBlocks.BAMBOO_TORCH_BLOCK_WALL, new Item.Settings().group(BambooTweaksItemGroups.BAMBOO_TWEAKS)));

	public BambooTweaksItems() {
		// NO-OP
	}

	public static void initialize() {

	}

	public static <T extends Item> T register(String name, T item) {
		return Registry.register(Registry.ITEM, BambooTweaks.identifier(name), item);
	}
}

File: common/src/main/kotlin/dev/vini2003/bambootweaks/common/registry/BambooTweaksItems.kt
package dev.vini2003.bambootweaks.common.registry

import me.shedaniel.architectury.registry.DeferredRegister
import me.shedaniel.architectury.registry.RegistrySupplier
import net.minecraft.item.Item
import net.minecraft.item.WallStandingBlockItem
import net.minecraft.util.registry.Registry
import dev.vini2003.bambootweaks.CommonBambooTweaks

object BambooTweaksItems {
	private val Items = BambooTweaksDeferredRegisters.Items
	
	fun init() {
	
	}
	
	fun <T : Item> register(name: String, supplier: () -> T): RegistrySupplier<T> {
		return Items.register(CommonBambooTweaks.identifier(name), supplier)
	}
	
	private fun itemSettings() = Item.Settings().group(BambooTweaksItemGroups.BambooTweaks)!!
	
	val BambooTorch = register("bamboo_torch") {
		WallStandingBlockItem(
			BambooTweaksBlocks.BambooTorch.get(),
			BambooTweaksBlocks.BambooWallTorch.get(),
			itemSettings()
		)
	}
	
	val DryBambooTorch = register("dry_bamboo_torch") {
		WallStandingBlockItem(
			BambooTweaksBlocks.DryBambooTorch.get(),
			BambooTweaksBlocks.DryBambooWallTorch.get(),
			itemSettings()
		)
	}
	
	val DryBamboo = register("dry_bamboo") {
		Item(itemSettings())
	}
}